src/deploy/functions/runtimes/python/index.ts

```diff
@@ -133,24 +133,30 @@ class Delegate implements runtimes.RuntimeDelegate {
     });
   }
 
-  async discoverSpec(
-    configValues: RuntimeConfigValues,
-    envs: EnvironmentVariables
-  ): Promise<Backend> {
+  async discoverBuild(
+    configValues: backend.RuntimeConfigValues,
+    envs: backend.EnvironmentVariables
+  ): Promise<Build> {
     let discovered = await discovery.detectFromYaml(this.sourceDir, this.projectId, this.runtime);
     if (!discovered) {
       const port = await portfinder.getPortPromise();
       const adminPort = await portfinder.getPortPromise({
         port: 8081,
       });
-      const killProcess = await this.serveAdmin(port, envs);
+      if (Object.keys(configValues || {}).length) {
+        envs.CLOUD_RUNTIME_CONFIG = JSON.stringify(configValues);
+      }
+      const processEnvs: backend.EnvironmentVariables = { ...envs };
+      if (Object.keys(configValues || {}).length) {
+        processEnvs.CLOUD_RUNTIME_CONFIG = JSON.stringify(configValues);
+      }
```

Comment:
Cloud Runtime Config is a feature only present for v1 runtimes (and unfortunately we're currently forced to support it for JS). We can cut it for v2 functions.